lib/logstash/outputs/scalyr.rb

```diff
@@ -991,6 +1004,46 @@ def add_client_timestamp_to_body(body)
     body[:client_timestamp] = current_time_millis.to_s
   end
 
+  # This should only be called on events that have already been determined to be too large
+  def truncate_event(event)
+    attrs = event[:attrs]
+    new_attrs = Hash.new
+    total_size = 0
+    priority_events = ["serverHost", "parser", "logfile", "severity", "message"]
+    priority_events.each do |key|
+      next unless attrs.has_key? key
+      total_size += key.bytesize
+      value = attrs.delete key
+      value_size = _get_size(value)
+      if value_size > @max_field_size_bytes
+        value.slice!(@max_field_size_bytes, value_size)
+        total_size += @max_field_size_bytes
+      else
+        total_size += value_size
+      end
+      new_attrs[key] = value
+    end
+    attrs.each do |key, value|
+      kv_size = key.bytesize + _get_size(value)
+      # skip field if the combined size of the key and value exceed the max field size.
```

Comment:
Yeah, I think this is a good call since doing that would be somewhat complex and possibly also add a lot of overhead during worst case scenario.

Having said that, I wonder if we still want to add some periodic opt-in debug logging of those offending fields to make troubleshooting easier for the end user?

Reply:
That makes sense.